### DISCUSSION TITLE
Regarding reading the elements of a TLE file.

### DISCUSSION THREAD

**Discussion by ajaypotdar99** [skyfielders/python-skyfield#820] [Q&A]
Hi,
Thanks for the code and the effort you have put in the project. It allows for an easier understanding and does make astrodynamical estimation convenient.
I happen to have a doubt regarding reading the entire text of a .tle file. For example, suppose I have a .tle file of a GPS satellite  ''tle-CATNR-32711.txt''. Now, what I want to do is read the file for the text such that the import of the two lines elements inside the .tle file is automatic in the code, and I don't have to read the .tle file in a string format first and then get line 1 and line 2 to input in EarthSatellite function for further processing.

I tried .tle_file using the following code:

```
n = 32711
url = 'https://celestrak.org/NORAD/elements/gp.php?CATNR={}'.format(n)
filename = 'tle-CATNR-{}.txt'.format(n)
satellite = load.tle_file(url, filename=filename)
```

, but all it does is give this as an output:
`'[<EarthSatellite NAVSTAR 62 (USA 201) catalog #32711 epoch 2022-12-10 19:10:30 UTC>]'`
which is just a list, and does not have the rest of the orbital information I want.

So, basically, is there a simpler way to do this, which is inherent to the package, that I seem to be missing, or is the reading of the .tle file similar to what is discussed in another one of your discussions: https://github.com/skyfielders/python-skyfield/discussions/677#discussion-3759784, in which he has a code bit, to parse the .tle file using `open(TLE_file, 'r')`?

Thanks in advance!

**Comment by brandon-rhodes** [comment]
> …which is just a list…

Yes, it's a list; but it's a list with a satellite object inside. If you call it, say, `sat_list` then you can say `sat_list[0]` to get the satellite itself with all its orbital information. Here's the Python Tutorial's introduction to how lists work, in case you need it:

https://docs.python.org/3/tutorial/introduction.html#lists

**Reply by ajaypotdar99** [reply]
Hi, yes, I do understand how lists work, perhaps in this case I am unable to grasp the satellite object working. Perhaps you could give another reference for that?
I did read about the EarthSatellite object, but then doesn't that require a different input of the the two TLE lines. Pardon me, but trying to piece together this info confused me.

**Reply by ajaypotdar99** [reply]
I read more and seemed to have figured it out. Thanks for the hint.

**Reply by brandon-rhodes** [reply]
I'm glad that the problem is resolved! Let me know if there is a phrase, or sentence, or paragraph that could be added to the documentation that would have prevented any confusion and that might make future users more productive.

**Reply by ajaypotdar99** [reply]
Yes, perhaps you could add about importing the orbital information in the 'Earth satellite' section of the document (https://rhodesmill.org/skyfield/earth-satellites.html#loading-a-tle-file). 
Maybe you could mention that importing the TLE data which is done using the `load.tle_file()` function gives an EarthSatellite object, the orbital information of which can be accessed using the `model` attribute, which you have mentioned in the Earth Satellite API reference. I am suggesting this because it didn't seem straightforward, and I didn't happen to find any examples either.

**Comment by Evidlo** [comment]
I found this issue when trying to figure out how to actually get access to the TLE data of a satellite object.  Since it wasn't mentioned above or anywhere in the docs, here is how to do that

``` python
>>> type(satellite)
skyfield.sgp4lib.EarthSatellite
>>> from sgp4 import exporter
>>> exporter.export_tle(satellite.model)
('1 25544U 98067A   25038.79531913  .00014753  00000-0  26690-3 0  9992',
 '2 25544  51.6383 227.9039 0003959 289.4821 164.9185 15.49898550495095')
```

If there's a better way to do this, let me know.
